services/siren/sql/recover_scenario.sql

```diff
@@ -1,7 +1,24 @@
-CREATE OR REPLACE FUNCTION recover_scenario(ScID bigint)
+CREATE OR REPLACE FUNCTION recover_scenario(ScID bigint, userId bigint, authRole text)
   RETURNS jsonb AS
 $func$
 BEGIN
+
+  IF (CASE authRole
+          WHEN 'Student' THEN (SELECT scenario.owner != userId FROM scenario WHERE scenario.id = ScID)
```

Comment:
In fact, I did not want to let students to delete scenarios at all; but probably you're right, let's keep it like this.